cmd/zpool/zpool_main.c

```diff
@@ -8449,21 +8469,28 @@ zpool_do_scrub(int argc, char **argv)
 
 	cb.cb_type = POOL_SCAN_SCRUB;
 	cb.cb_scrub_cmd = POOL_SCRUB_NORMAL;
+	cb.cb_date_start = cb.cb_date_end = 0;
 
 	boolean_t is_error_scrub = B_FALSE;
 	boolean_t is_pause = B_FALSE;
 	boolean_t is_stop = B_FALSE;
 	boolean_t is_txg_continue = B_FALSE;
 
 	/* check options */
-	while ((c = getopt(argc, argv, "spweC")) != -1) {
+	while ((c = getopt(argc, argv, "spweCE:S:")) != -1) {
 		switch (c) {
 		case 'e':
 			is_error_scrub = B_TRUE;
 			break;
+		case 'E':
+			cb.cb_date_end = date_string_to_sec(optarg);
```

Comment:
I wonder if here we should round to the end of the minute.  Because right now it might depend on user doing the rounding up to reliably cover the wanted range.

Reply:
It looks more consistent now.  But I can't guess when `ROUNDING_DOWN` would make sense there.

Reply:
Ok, I have get rid of it.